module/zfs/dnode.c

```diff
@@ -1778,15 +1778,24 @@ dnode_try_claim(objset_t *os, uint64_t object, int slots)
 }
 
 /*
- * Checks if the dnode contains any uncommitted dirty records.
+ * Checks if the dnode itself is dirty, or is carrying any uncommitted records.
```

Comment:
any uncommitted records --->
any uncommitted record

Reply:
I'm not sure that the change you suggest is right, actually. I think it's fine as it is.

Reply:
Nah, in this case it really is "records", that is, `dbuf_dirty_record_t` objects attached to `dn_dirty_list`.